src/pytorch_lightning/cli.py

```diff
@@ -550,9 +551,17 @@ def _instantiate_trainer(self, config: Dict[str, Any], callbacks: List[Callback]
                 value = self.trainer_defaults[key]
                 config[key] += value if isinstance(value, list) else [value]
             if self.save_config_callback and not config.get("fast_dev_run", False):
+                to_save = self.config.get(str(self.subcommand), self.config)
+                # TODO: remove this override in v2.0.0
+                if (
+                    "trainer" in to_save
+                    and "logger" in to_save.trainer
+                    and isinstance(to_save.trainer.logger, _NoneSentinel)
+                ):
+                    to_save.trainer.logger = None
```

Comment:
@mauvilsa To carry out this deprecation, I had to write a None-like sentinel value to know if the user passed something.

With this hack here, I avoid saving the sentinel in configs.

However, I need to do the same for when `--print_config` is called. I haven't easily found a way to do that.

Can you help? Thank you!

Reply:
You just need to register `_NoneSentinel` as a type that serializes to `None`, i.e.,
```python
from jsonargparse.typing import register_type
register_type(_NoneSentinel, serializer=lambda v: None)
```
No need for this custom saving.